runners/pom.xml

```diff
@@ -55,6 +55,13 @@
             <plugin>
               <groupId>org.apache.maven.plugins</groupId>
               <artifactId>maven-surefire-plugin</artifactId>
+              <dependencies>
+                <dependency>
+                  <groupId>org.apache.beam</groupId>
+                  <artifactId>beam-sdks-java-core</artifactId>
+                  <type>test-jar</type>
```

Comment:
I see. Yes, tests of APIs that are not public need to be adjusted alongside the classes under test. I would actually expect that ROS tests don't do this.

But indeed, the main test-jar should match the main output jar. Users of runnable-on-service tests shouldn't have to reason about how the jar is produced. I'm sure we can make that work.